src/mito2/src/engine.rs

```diff
@@ -28,7 +28,8 @@ mod create_test;
 mod drop_test;
 #[cfg(test)]
 mod flush_test;
-#[cfg(test)]
+#[cfg(any(test, feature = "test"))]
+#[cfg_attr(feature = "test", allow(unused))]
 pub(crate) mod listener;
```

Comment:
Do we still need `allow(unused)` if we pub this mod? It's fine to pub it under the test feature.